src/conda_devenv/devenv.py

```diff
@@ -110,23 +220,15 @@ def render_jinja(contents: str, filename: Path, is_included: bool) -> str:
         "root": os.path.dirname(os.path.abspath(filename)),
         "sys": sys,
         "aarch64": "aarch64" == platform.machine(),
-        "arm64": isosx and "arm64" == platform.machine(),
+        "arm64": conda_platform.name == "osx" and "arm64" == platform.machine(),
         "x86": "x86" == platform.machine(),
         "x86_64": "x86_64" == platform.machine(),
```

Comment:
I was a bit unsure, because we change the `CondaPlatform` when rendering for different platforms other than the host, but those settings would probably be incorrect because `platform.machine()` is not affected.

What should we do?

Reply:
We can find which flags in the form `# [arm64]` are being used, however we cannot be 100% sure because the flag could be used inside a Jinja2 expression.

How about we leave a note in the docs that those platforms are not safe to use with lock files? At least as a stop gap measure for now.

Reply:
OK. Let it this way.
I think the code will crash if someone try to use this for restricting the platform using the lock infra.

Reply:
Added a note at the end of the docs for locking. 👍